contracts/test/StakeTable.t.sol

```diff
@@ -0,0 +1,225 @@
+    uint64 depositAmount;
+    uint64 validUntilEpoch;
+
+    function computeGroupElements(address sender)
+        private
+        returns (BN254.G2Point memory, EdOnBN254.EdOnBN254Point memory, BN254.G1Point memory)
+    {
+        depositAmount = 10;
+        validUntilEpoch = 5;
+
+        // Generate a BLS signature and other values using rust code
+        string[] memory cmds = new string[](3);
+        cmds[0] = "diff-test";
+        cmds[1] = "gen-bls-sig";
+        cmds[2] = vm.toString(sender);
+
+        bytes memory result = vm.ffi(cmds);
+        (
+            uint256 blsSigX,
+            uint256 blsSigY,
+            uint256 blsVKx0,
+            uint256 blsVKx1,
+            uint256 blsVKy0,
+            uint256 blsVKy1,
+            uint256 schnorrVKx,
+            uint256 schnorrVKy
+        ) = abi.decode(
+            result, (uint256, uint256, uint256, uint256, uint256, uint256, uint256, uint256)
+        );
+
+        return (
+            BN254.G2Point(blsVKx1, blsVKx0, blsVKy1, blsVKy0), // blsVK. Note: (x,y) coordinates for
+                // each field component must be inverted.
+            EdOnBN254.EdOnBN254Point(schnorrVKx, schnorrVKy), // schnorrVK
+            BN254.G1Point(blsSigX, blsSigY) // sig
+        );
+    }
+
+    function setUp() public {
+        exampleTokenCreator = msg.sender;
+        vm.prank(exampleTokenCreator);
+        token = new ExampleToken(INITIAL_BALANCE);
+        tokenAddress = address(token);
+
+        LightClient.LightClientState memory genesis = LightClient.LightClientState({
+            viewNum: 0,
+            blockHeight: 0,
+            blockCommRoot: 0,
+            feeLedgerComm: 0,
+            stakeTableBlsKeyComm: 0,
+            stakeTableSchnorrKeyComm: 0,
+            stakeTableAmountComm: 0,
+            threshold: 0
+        });
+        lightClientContract = new LightClient(genesis,10);
+        lightClientAddress = address(lightClientContract);
+
+        stakeTable = new S(tokenAddress,lightClientAddress);
+    }
+
+    /// @dev Tests a correct key registation
+    function testRegisterHappyPath() external {
+        address msgSenderAddress = msg.sender;
+
+        (
+            BN254.G2Point memory blsVK,
+            EdOnBN254.EdOnBN254Point memory schnorrVK,
+            BN254.G1Point memory sig
+        ) = computeGroupElements(msgSenderAddress);
+
+        // Prepare for the token transfer
+        vm.prank(msg.sender);
+        token.approve(address(stakeTable), depositAmount);
+
+        vm.prank(msgSenderAddress);
+
+        // Balances before registration
+        assertEq(token.balanceOf(msgSenderAddress), INITIAL_BALANCE);
+
+        uint256 nativeAmount;
+        uint256 restakedAmount;
+        (nativeAmount, restakedAmount) = stakeTable.totalStake();
+        assertEq(nativeAmount, 0);
+        assertEq(restakedAmount, 0);
+
+        // Check event is emitted
+        vm.expectEmit(false, false, false, true, address(stakeTable));
```

Comment:
try to declare `expectEmit/Revert` 1 line above the actual contract invocation for readability.

Reply:
Put vm.expectEmit closer to the call to contract in 1d85c9f, even though it looks like the expected event must be defined before the call.